yyyymmdd/range.py

```diff
@@ -351,9 +382,174 @@ def stop_max_inv(x, y):
             gap = offset - crt
             filler = gap if 0 == gap % step else (gap // step + 1) * step
         start = self.start + crt + filler
-        stop = stop_min(self.stop, other.stop) if sign > 0 else stop_max_inv(self.stop, other.stop)
+        stop = (
+            stop_min(self.stop, other.stop)
+            if sign > 0
+            else stop_max_inv(self.stop, other.stop)
+        )
         return type(self)(start, stop, step)
 
+    # ===========================================================================
+    # mereological part structure
+    # ===========================================================================
+
+    def __lt__(self, other):
+        return self.is_proper_part(other)
+
+    def __gt__(self, other):
+        return other.is_proper_part(self)
+
+    def __le__(self, other):
+        if self.step != 1 or other.step != 1:
+            raise ValueError(
```

Comment:
I wonder if `NotImplementedError` is more suitable here, because for (at least) some of the mereology operations, we _could_ implement them.
I'm fine with leaving it unimplemented and raising `NotImplementedError`, until someone decides it is worth the effort to do the extra work.